loom/common/core/allocator.h

```diff
@@ -254,6 +175,25 @@ struct loom_allocator
     loom_allocator_t            *parent;
 };
 
+// Allocation callback function pointers
+typedef void (*loom_allocator_callback_free_t)(loom_allocator_t *thiz, void *inner, size_t size, const char *file, int line);
+
+typedef struct loom_debugAllocatorCallbacks loom_debugAllocatorCallbacks_t;
+struct loom_debugAllocatorCallbacks
+{
+    loom_allocator_callback_free_t onFree;
+    loom_debugAllocatorCallbacks_t* next;
+};
+
+// Register allocation function callbacks struct
```

Comment:
Yeah, I gave up in the middle of writing that comment apparently, lol. Will fix.